packages/preset-icons/src/types.ts

```diff
@@ -81,8 +92,14 @@ export interface IconsOptions {
    * - https://cdn.skypack.dev/
    */
   cdn?: string
+
   /**
    * Custom fetch function to provide the icon data.
    */
   customFetch?: (url: string) => Promise<any>
+
+  /**
+   * Custom the final additional props.
+   */
+  propsProcessor?: (props: Record<string, string>, collection?: string, icon?: string, svg?: string, mode?: string) => void
```

Comment:
I wonder if we could have a more general purpose `processor` that could modify more than the props, but possibly also the `props`, `svg`, `cssObject`?

Reply:
I fixed it and provided more context for users to use.